runtime/src/accounts_db.rs

```diff
@@ -1351,6 +1387,50 @@ impl AccountsDB {
         hasher.result()
     }
 
+    pub fn blake3_hash_account_data(
+        slot: Slot,
+        lamports: u64,
+        owner: &Pubkey,
+        executable: bool,
+        rent_epoch: Epoch,
+        data: &[u8],
+        pubkey: &Pubkey,
+    ) -> Hash {
+        if lamports == 0 {
+            return Hash::default();
+        }
+
+        let mut hasher = blake3::Hasher::new();
+
+        hasher.update(&lamports.to_le_bytes());
+
+        hasher.update(&slot.to_le_bytes());
+
+        hasher.update(&rent_epoch.to_le_bytes());
+
+        hasher.update(&data);
+
+        if executable {
+            hasher.update(&[1u8; 1]);
+        } else {
+            hasher.update(&[0u8; 1]);
+        }
+
+        hasher.update(&owner.as_ref());
+        hasher.update(&pubkey.as_ref());
+
+        Hash(<[u8; solana_sdk::hash::HASH_BYTES]>::try_from(hasher.finalize().as_slice()).unwrap())
```

Comment:
This is a little scary.  We already do no domain separation of our SHA256 usage, sticking Blake3 behind `Hash` seems like asking for trouble. It feels to me like `Hash` should be a trait, but I'll expand on that in an issue.

I'd feel better if this was a separate `AccountHash` type.  Unfortunately that'll make the migration a bit tricky.  The new type will have to be able to temporarily carry a `Hash` as well

cc/ @sakridge

Reply:
Yea, I would prefer a separate type as well, but I'm not sure how involved that is. I think using this for now is acceptable.